### ISSUE TITLE
How can I create rec data from my numpy arrays?

### ISSUE THREAD

**Issue by metorm** [apache/incubator-mxnet#5204]
I get quite a large data set (~100G) to train a model.
Each data item is a `[2,256,64]` numpy array contains float number. Looks much like an image with two channels.
Discussion about rec are all about converting images to rec. Anyone tell me how to convert my numpy array to rec format?

**Comment by saicoco** [comment]
You can refer the im2rec.py and replace the image arrar opencv read with you data, I guess it will be ok. 在2017年03月02日 11:17，Wesley Ranger 写道： I get quite a large data set (~100G) to train a model. Each data item is a [2,256,64] numpy array contains float number. Looks much like an image with two channels. Discussion about rec are all about converting images to rec. Anyone tell me how to convert my numpy array to rec format? — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by lincolnBush** [comment]
@metorm @saicoco, this might not work as there is a image compression step there. Have you come up with other ways to solve it?

**Comment by metorm** [comment]
I think you can turn off compression. However, my array is of float type, and converting it must causes accurancy loss. I am writing a data base adapter for this. I'll post it if the adapter works.



发自我的小米手机
在 lincolnBush <notifications@github.com>，2017年3月2日 17:55写道：

@metorm<https://github.com/metorm> @saicoco<https://github.com/saicoco>, this might not work as there is a image compression step there. Have you come up with other ways to solve it?

**Comment by metorm** [comment]
@lincolnBush
Here is a simple version of my own adapter. Just a toy to fit a sine() function:

```
import mxnet as mx
import numpy as np
import logging
import sys
import os
from mxnet.io import DataIter, DataBatch


class TestIterater(DataIter):

    def __init__(self,
                 data_name="data",
                 label_name="softmax_label",
                 batch_size=50):
        super(TestIterater, self).__init__()
        self.data_name = data_name
        self.label_name = label_name

        self.num_data = 100000
        self.cursor = 0
        self.batch_size = batch_size

    @property
    def provide_data(self):
        return [(self.data_name, (1, 1, 1))]

    @property
    def provide_label(self):
        return [(self.label_name, (1,))]

    def get_batch_size(self):
        return self.batch_size

    def reset(self):
        self.cursor = 0

    def iter_next(self):
        return self.cursor < self.num_data

    def next(self):
        if self.iter_next():
            self.cursor += self.batch_size
            x = np.random.uniform(0, 2 * np.pi, (self.get_batch_size(), 1, 1, 1))
            y = np.sin(x)
            y = y.reshape(self.get_batch_size(), 1)
            return DataBatch(
                data=[mx.nd.array(v) for v in x],
                label=[mx.nd.array(v) for v in y]
            )
        else:
            raise StopIteration

if __name__ == '__main__':
    train_epoch = 10
    batch_size = 5
    train_iter = TestIterater(batch_size=50)
    test_iter = TestIterater(batch_size=50)

    net_in = mx.symbol.Variable('data')
    net_in_flat = mx.symbol.Flatten(data=net_in)

    net_label = mx.symbol.Variable('softmax_label')

    fc1 = mx.symbol.FullyConnected(data=net_in_flat, name='fc1', num_hidden=512)
    # fc1_act = mx.symbol.Activation(data=fc1, name='fc1_act', act_type='sigmoid')
    fc2 = mx.symbol.FullyConnected(data=fc1, name='fc2', num_hidden=256)
    # fc2_act = mx.symbol.Activation(data=fc2, name='fc2_act', act_type='sigmoid')
    fc3 = mx.symbol.FullyConnected(data=fc2, name='fc3', num_hidden=256)
    net_out = mx.symbol.FullyConnected(data=fc3, name='net_out', num_hidden=1)
    loss = mx.symbol.LinearRegressionOutput(data=net_out, label=net_label, name='loss')

    logging.basicConfig(level=logging.INFO)
    logger = logging.getLogger(__name__)  # get a logger to accuracies are printed
    model = mx.module.Module(
        symbol=loss,
        data_names=['data'],
        label_names=['softmax_label'],
        logger=logger
    )

    model.fit(
        train_data=train_iter,
        eval_data=test_iter,
        eval_metric='mae',
        optimizer='sgd',
        # optimizer_params=(('learning_rate', 0.005), ('momentum', 0.5)),
        num_epoch=train_epoch,
        batch_end_callback=mx.callback.Speedometer(batch_size, frequent=100)
    )
```

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to ping me to reopen if this is still an active issue. Thanks!
